### PR TITLE
zh: TLS for MySQL Client

### PR DESCRIPTION
**Pull request by weekface**
### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v1.1 changes for now)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s):

### CODE REVIEW

File: zh/enable-tls-for-mysql-client.md [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+title: 为 MySQL 客户端开启 TLS
+summary: 在 Kubernetes 上如何为 TiDB 集群的 MySQL 客户端开启 TLS。
+category: how-to
+---
+
+# 为 MySQL 客户端开启 TLS
+
+本文主要描述了在 Kubernetes 上如何为 TiDB 集群的 MySQL 客户端开启 TLS。TiDB Operator 从 v1.1 开始已经支持为 Kubernetes 上 TiDB 集群开启 MySQL 客户端 TLS。开启步骤为：
+
+1. 为 TiDB Server 颁发一套 Server 端证书，为 MySQL Client 颁发一套 Client 端证书。并创建两个 Secret 对象，Secret 名字分别为：`<cluster-name>-tidb-server-secret` 和  `<cluster-name>-tidb-client-secret`，分别包含前面创建的两套证书;
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]:
```suggestion
1. 为 TiDB Server 颁发一套 Server 端证书，为 MySQL Client 颁发一套 Client 端证书。并创建两个 Secret 对象，Secret 名字分别为：`<cluster-name>-tidb-server-secret` 和  `<cluster-name>-tidb-client-secret`，分别包含前面创建的两套证书；
```

File: zh/enable-tls-for-mysql-client.md [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+2. 部署集群，设置 `.spec.tidb.tlsClient.enabled` 属性为 `true`；
+3. MySQL 客户端使用方法。
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]: The 3rd step does not look like a step. Please rewrite it.

File: zh/enable-tls-for-mysql-client.md [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+其中，颁发证书的方式有多种，本文档提供两种方式，用户也可以根据需要为 TiDB 集群颁发证书，这两种方式分别为：
+
+- 使用 `cfssl` 系统颁发证书；
+- 使用 `cert-manager` 系统颁发证书；
+
+## 第一步：为 TiDB 集群颁发两套证书
+
+### 使用 `cfssl` 系统颁发证书
+
+1. 首先下载 `cfssl` 软件并初始化证书颁发机构
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]:
```suggestion
1. 首先下载 `cfssl` 软件并初始化证书颁发机构：
```

File: zh/enable-tls-for-mysql-client.md [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+    chmod +x ~/bin/{cfssl,cfssljson}
+    export PATH=$PATH:~/bin
+    
+    mkdir -p ~/cfssl
+    cd ~/cfssl
+    cfssl print-defaults config > ca-config.json
+    cfssl print-defaults csr > ca-csr.json
+    ```
+
+2. 在 `ca-config.json` 配置文件中配置 CA 选项
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]:
```suggestion
2. 在 `ca-config.json` 配置文件中配置 CA 选项：
```

File: zh/enable-tls-for-mysql-client.md [hunk 5]
```diff
@@ -0,0 +1,448 @@
...
+                        "key encipherment",
+                        "client auth"
+                    ]
+                }
+            }
+        }
+    }
+    ```
+
+3. 您还可以修改 `ca-csr.json` 证书签名请求（CSR）
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]:
```suggestion
3. 您还可以修改 `ca-csr.json` 证书签名请求 (CSR)：
```

File: zh/enable-tls-for-mysql-client.md [hunk 6]
```diff
@@ -0,0 +1,448 @@
...
+                "L": "CA",
+                "O": "PingCAP",
+                "ST": "Beijing",
+                "OU": "TiDB"
+            }
+        ]
+    }
+    ```
+
+4. 使用定义的选项生成 CA
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]:
```suggestion
4. 使用定义的选项生成 CA：
```

File: zh/enable-tls-for-mysql-client.md [hunk 7]
```diff
@@ -0,0 +1,448 @@
...
+
+    {{< copyable "shell-regular" >}}
+    
+    ``` shell
+    cfssl gencert -initca ca-csr.json | cfssljson -bare ca -
+    ```
+
+5. 生成 Server 端证书
+
+    首先生成默认的 `server.json` 文件：
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]:
```suggestion
5. 生成 Server 端证书。

    首先生成默认的 `server.json` 文件：
```

File: zh/enable-tls-for-mysql-client.md [hunk 8]
```diff
@@ -0,0 +1,448 @@
...
+    
+    最后生成 Server 端证书：
+    
+    {{< copyable "shell-regular" >}}
+    
+    ``` shell
+    cfssl gencert -ca=ca.pem -ca-key=ca-key.pem -config=ca-config.json -profile=server server.json | cfssljson -bare server
+    ```
+
+6. 生成 Client 端证书
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]:
```suggestion
6. 生成 Client 端证书：
```

Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]:
```suggestion
6. 生成 Client 端证书。
```

File: zh/enable-tls-for-mysql-client.md [hunk 9]
```diff
@@ -0,0 +1,448 @@
...
+    
+    最后生成 Client 端证书：
+    
+    {{< copyable "shell-regular" >}}
+    
+    ``` shell
+    cfssl gencert -ca=ca.pem -ca-key=ca-key.pem -config=ca-config.json -profile=client client.json | cfssljson -bare client
+    ```
+
+7. 创建 Kubernetes Secret 对象
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]:
```suggestion
7. 创建 Kubernetes Secret 对象。
```

File: zh/enable-tls-for-mysql-client.md [hunk 10]
```diff
@@ -0,0 +1,448 @@
...
+    kubectl create secret generic <cluster-name>-tidb-client-secret --namespace=<namespace> --from-file=tls.crt=~/cfssl/client.pem --from-file=tls.key=~/cfssl/client-key.pem --from-file=ca.crt=~/cfssl/ca.pem
+    ```
+
+    这样就给 Server/Client 端证书分别创建了：
+
+    - 一个 Secret 供 TiDB Server 启动时加载使用;
+    - 另一个 Secret 供 MySQL 客户端连接 TiDB 集群时候使用。
+
+用户可以生成多套 Client 端证书，并且至少要生成一套 Client 证书供 TiDB Operator 内部组件访问 TiDB Server（目前有 TidbInitializer 会访问 TiDB
+Server 来设置密码或者一些初始化操作）。
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]: 不用换行哈。

File: zh/enable-tls-for-mysql-client.md [hunk 11]
```diff
@@ -0,0 +1,448 @@
...
+
+### 使用 cert-manager 颁发证书
+
+1. 安装 cert-manager
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]:
```suggestion
1. 安装 cert-manager。
```

File: zh/enable-tls-for-mysql-client.md [hunk 12]
```diff
@@ -0,0 +1,448 @@
...
+
+    请参考官网安装：[cert-manager installation in Kubernetes](https://docs.cert-manager.io/en/release-0.11/getting-started/install/kubernetes.html)。
+
+2. 创建一个 ClusterIssuer 用于给 TiDB 集群颁发证书
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]:
```suggestion
2. 创建一个 ClusterIssuer 用于给 TiDB 集群颁发证书。
```

File: zh/enable-tls-for-mysql-client.md [hunk 13]
```diff
@@ -0,0 +1,448 @@
...
+      name: tidb-server-issuer
+    spec:
+      ca:
+        secretName: tidb-server-issuer-cert
+    ```
+
+    上面的文件创建三个对象：
+
+    - 一个 SelfSigned 类型的 ClusterIsser 对象（用于生成CA 类型 ClusterIssuer 所需要的 CA 证书）;
+    - 一个 Cerfificate 对象，`isCa` 属性设置为 `true`；
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]:
```suggestion
    - 一个 SelfSigned 类型的 ClusterIsser 对象（用于生成 CA 类型 ClusterIssuer 所需要的 CA 证书）;
    - 一个 Certificate 对象，`isCa` 属性设置为 `true`；
```

File: zh/enable-tls-for-mysql-client.md [hunk 14]
```diff
@@ -0,0 +1,448 @@
...
+
+    最后执行下面的命令进行创建：
+
+    {{< copyable "shell-regular" >}}
+
+    ``` shell
+    kubectl apply -f ~/cert-manager/tidb-server-issuer.yaml
+    ```
+
+3. 创建 Server 端证书
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]:
```suggestion
3. 创建 Server 端证书。
```

File: zh/enable-tls-for-mysql-client.md [hunk 15]
```diff
@@ -0,0 +1,448 @@
...
+
+    {{< copyable "shell-regular" >}}
+
+    ``` shell
+    kubectl apply -f ~/cert-manager/tidb-server-cert.yaml
+    ```
+
+    创建这个对象以后，cert-manager 会生成一个名字为 `<cluster-name>-tidb-server-secret` 的 Secret 对象供 TiDB Server 使用。
+
+4. 创建 Client 端证书
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]:
```suggestion
4. 创建 Client 端证书。
```

File: zh/enable-tls-for-mysql-client.md [hunk 16]
```diff
@@ -0,0 +1,448 @@
...
+
+    然后来创建 Client 端证书，创建一个 `tidb-client-cert.yaml` 文件，并输入以下内容：
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]:
```suggestion
    创建一个 `tidb-client-cert.yaml` 文件，并输入以下内容：
```

File: zh/enable-tls-for-mysql-client.md [hunk 17]
```diff
@@ -0,0 +1,448 @@
...
+      - `<cluster-name>-tidb.<namespace>.svc`
+      - `*.<cluster-name>-tidb`
+      - `*.<cluster-name>-tidb.<namespace>`
+      - `*.<cluster-name>-tidb.<namespace>.svc`
+    - `ipAddresses` 需要填写这两个 IP ，根据需要可以填写其他 IP：
+      - `127.0.0.1`
+      - `::1`
+    - `issuerRef` 请填写上面创建的 ClusterIssuer；
+    - 其他属性请参考
+      [cert-manager API](https://cert-manager.io/docs/reference/api-docs/#cert-manager.io/v1alpha2.CertificateSpec)。
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]:
```suggestion
    - 其他属性请参考 [cert-manager API](https://cert-manager.io/docs/reference/api-docs/#cert-manager.io/v1alpha2.CertificateSpec)。
```

File: zh/enable-tls-for-mysql-client.md [hunk 18]
```diff
@@ -0,0 +1,448 @@
...
+      ipAddresses:
+        - 127.0.0.1
+        - ::1
+      issuerRef:
+        name: tidb-server-issuer
+        kind: ClusterIssuer
+        group: cert-manager.io
+    ```
+
+    其中 `<cluster-name>` 为集群的名字
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]:
```suggestion
    其中 `<cluster-name>` 为集群的名字：
```

File: zh/enable-tls-for-mysql-client.md [hunk 19]
```diff
@@ -0,0 +1,448 @@
...
+      usages:
+        - client auth
+      issuerRef:
+        name: tidb-server-issuer
+        kind: ClusterIssuer
+        group: cert-manager.io
+    ```
+
+    其中 `<cluster-name>` 为集群的名字：
+    - `spec.secretName` 请设置为 `<cluster-name>-tidb-client-secret`；
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]:
```suggestion
    其中 `<cluster-name>` 为集群的名字：
    
    - `spec.secretName` 请设置为 `<cluster-name>-tidb-client-secret`；
```

File: zh/enable-tls-for-mysql-client.md [hunk 20]
```diff
@@ -0,0 +1,448 @@
...
+    - `usages` 请添加上 `client auth`；
+    - `dnsNames` 和 `ipAddresses` 不需要填写；
+    - `issuerRef` 请填写上面创建的 ClusterIssuer；
+    - 其他属性请参考
+      [cert-manager API](https://cert-manager.io/docs/reference/api-docs/#cert-manager.io/v1alpha2.CertificateSpec)。
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]:
```suggestion
    - 其他属性请参考 [cert-manager API](https://cert-manager.io/docs/reference/api-docs/#cert-manager.io/v1alpha2.CertificateSpec)。
```

File: zh/enable-tls-for-mysql-client.md [hunk 21]
```diff
@@ -0,0 +1,448 @@
...
+    {{< copyable "shell-regular" >}}
+
+    ``` shell
+    kubectl apply -f ~/cert-manager/tidb-client-cert.yaml
+    ```
+
+    创建这个对象以后，cert-manager 会生成一个名字为 `<cluster-name>-tidb-client-secret` 的 Secret 对象供 TiDB Client 使用。
+
+用户可以生成多套 Client 端证书，并且至少要生成一套 Client 证书供 TiDB Operator 内部组件访问 TiDB Server（目前有 TidbInitializer 会访问 TiDB
+Server 来设置密码或者一些初始化操作）。
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]: ditto

File: zh/enable-tls-for-mysql-client.md [hunk 22]
```diff
@@ -0,0 +1,448 @@
...
+
+## 第二步：部署 TiDB 集群
+
+接下来将会通过两个 CR 对象来创建一个 TiDB 集群，并且：
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]: 冒号前尽量使用完整的句子。
```suggestion
接下来将会通过两个 CR 对象来创建一个 TiDB 集群，并且执行以下步骤：
```

File: zh/enable-tls-for-mysql-client.md [hunk 23]
```diff
@@ -0,0 +1,448 @@
...
+
+其中 `<cluster-name>` 为集群的名字，`<namespace>` 为 TiDB 集群部署的命名空间。通过设置 `spec.tidb.tlsClient.enabled` 属性为 `true` 来开启 MySQL 客户端 TLS。
+
+将上面文件保存为 `cr.yaml`，然后使用 `kubectl apply -f cr.yaml` 来创建 TiDB 集群。
+
+## 第三步：MySQL 客户端使用方法
+
+可以根据
+[官网文档](https://github.com/pingcap/tidb-operator/blob/master/docs/CONTRIBUTING.md#start-tidb-operator-locally-and-do-manual-tests)
+提示，使用上面创建的 Client 证书，通过下面的方法连接 TiDB 集群：
```
Comment (ran-huang) [inline zh/enable-tls-for-mysql-client.md]: ditto

File: zh/enable-tls-for-mysql-client.md (hunk 7 above)
Comment (weekface) [reply zh/enable-tls-for-mysql-client.md]: 这里要用句号吗？

**Comment by weekface** [comment]
@ran-huang Comments addressed PTAL.

File: zh/enable-tls-for-mysql-client.md (hunk 7 above)
Comment (ran-huang) [reply zh/enable-tls-for-mysql-client.md]: > 这里要用句号吗？

句号或冒号都可以，但同一篇文档内用法最好统一。请检查一下本篇文档各个步骤的标点是否一致~

**Comment by ran-huang** [comment]
LGTM
@DanielZhangQD PTAL, thanks!

**Pull request merged**